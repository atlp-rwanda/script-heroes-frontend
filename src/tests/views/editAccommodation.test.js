import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import UpdateAccomodationsPage from "../../components/accommodation/editAccommodation";

jest.mock("../../components/accommodation/editAccommodation", () => {
  return {
    __esModule: true,
    default: () => {
      return <div></div>;
    },
  };
});

describe("edit accommodation component", () => {
  const initialState = {
    accommodation: { accomodationsObj: [] },
    locations: { locations: [] },
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  let store;

  it("Should render Edit Accommodation page", () => {
    store = mockStore(initialState);
    let wrapper = mount(
      <Provider store={store}>
        <Router>
          <UpdateAccomodationsPage />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
