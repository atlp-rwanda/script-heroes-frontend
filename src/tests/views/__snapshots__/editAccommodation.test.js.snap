// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit accommodation component Should render Edit Accommodation page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <default>
        <div />
      </default>
    </Router>
  </BrowserRouter>
</Provider>
`;
